platforms/avr/platform.mk

```
@@ -30,6 +30,21 @@ CXXFLAGS += -fno-exceptions -std=c++11
 
 LDFLAGS +=-Wl,--gc-sections
 
+# Force linking of AVR's libc minimal printf implementation which
+# can shave ~400 bytes. Usually we use the xprintf implementation
+# but keyboards that use s(n)printf automatically pull in the AVR
+# libc implementation, which is ~900 bytes heavy.
+# Unfortunately the mechanism to use the minimal implementation is
+# to forcefully mark vfprintf as undefined, which in turn forces
+# the linker to ALWAYS include the library in the final binary.
+# This is even done when the function is not used, so this has
+# to be done on a individual per keyboard level.
+# See http://savannah.nongnu.org/bugs/?50811
+AVR_FORCE_MINIMAL_PRINTF_IMPLEMENTATION ?= no
+ifeq ($(strip $(AVR_FORCE_MINIMAL_PRINTF_IMPLEMENTATION)), yes)
+	LDFLAGS += -Wl,-u,vfprintf -lprintf_min
```

Apparently there is another way to perform the printf implementation replacement that does not increase the firmware size even if `vfprintf()` is not actually used:
```suggestion
	LDFLAGS += -Wl,--whole-archive -lprintf_min -Wl,--no-whole-archive
```
(`-Wl,--whole-archive` forces the whole contents of the library to be included, as if it was an object file; however, `-Wl,--gc-sections` is still able to discard unused functions if there is no `-Wl,-u,vfprintf`).

---

Oh great, so this can become the default behavior! There is also a minimal scanf implementation. Usually there is no need for it on QMK but if it gets thrown out we can include it regardless. I'll test and push that fix then.

---

Alright, mentioned the limitations and updated the code. Also set you as an co-author, thanks!